### PR TITLE
fix(core): Canceling the job doesn't cancel the actual process

### PR DESCRIPTION
**Pull request by MohdRafiUddin**
* fix(core): Canceling the job doesn't cancel the actual process.

### CODE REVIEW

File: src/core/removeCommonResource.js [hunk 1]
```diff
@@ -1,35 +1,58 @@
...
+  await forEachObjIndexed(async (vdr) => {
     try {
       if (isJobCancelled(jobId)) {
-          removeCancelledJobId(jobId);
-          throw new Error('job is cancelled');
+        emitter.emit(EventTopic.ASSET_STATUS, {
+          processId,
+          assetType: Assets.VDRS,
+          assetName: vdr.vdrName,
+          assetStatus: ArtifactStatus.CANCELLED,
```
Comment (GiriNellore) [inline src/core/removeCommonResource.js:25]: **question:** Why emit this cancelled event from here when this event is already emitted from service? This becomes a circular event emit, and IMO redundant.

File: src/core/removeElement.js [hunk 2]
```diff
@@ -1,35 +1,59 @@
...
+    return;
+  }
+  const removePromises = await elements.map(async (element) => {
+    try {
+      if (isJobCancelled(jobId)) {
+        emitter.emit(EventTopic.ASSET_STATUS, {
+          processId,
+          assetType: Assets.ELEMENTS,
+          assetName: element.key,
+          assetStatus: ArtifactStatus.CANCELLED,
```
Comment (GiriNellore) [inline src/core/removeElement.js:25]: **question:** same question as above

**Review by GiriNellore — CHANGES_REQUESTED** [review]
@MohdRafiUddin Left few comments

File: src/core/removeCommonResource.js (hunk 1 above)
Comment (MohdRafiUddin) [reply src/core/removeCommonResource.js:25]: This is an asset update event.

File: src/events/emitter.js [hunk 3]
```diff
@@ -11,8 +11,9 @@ const singleton = () => {
 
 const EventTopic = {
   ASSET_STATUS: 'ASSERT_STATUS',
-  JOB_CANCELLED: 'JOB_CANCELLED',
   JOB_CREATED: 'JOB_CREATED',
+  JOB_CANCELLED_CORE: 'JOB_CANCELLED_CORE',
```
Comment (GiriNellore) [inline src/events/emitter.js]: **nitpick:** remove this unused item

File: src/events/cancelled-job.js [hunk 4]
```diff
@@ -1,30 +1,39 @@
...
-    const idSet = getIdSet();
+  if (!jobId) {
+    return;
+  }
+  const idSet = getIdSet();
+  console.log('isJobCancelled - Content of the Cancelled Job Id Queue', idSet);
+  if (idSet.has(jobId)) {
     return idSet.has(jobId);
-}
+  } else {
```
Comment (GiriNellore) [inline src/events/cancelled-job.js]: **nitpick:** remove this unused loop

**Review by GiriNellore — COMMENTED** [review]
@MohdRafiUddin Looks good except for minor nitpicks.

**Review by GiriNellore — APPROVED** [review]

**Pull request merged**